### PR TITLE
0.16.0 review - opened for coderabbit review

### PR DESCRIPTION
**Pull request by phatchman**
Upgrade to Zig 0.16.0.
Support CDOS
Various FIXES.

## Summary by CodeRabbit

* **New Features**
  * Added disk label commands (get/set) for disk images (including CDOS label handling), with GUI support updates.
  * Expanded supported CDOS disk variants and improved format autodetection behavior.
  * GUI bumped to 0.10.0 with refreshed UI and a new dark terminal theme.
* **Documentation**
  * Added LLM/AI disclosure and policy, plus quick-start links and expanded CDOS/supported-type guidance.
* **Refactor**
  * Reworked disk/image and command I/O flows for unified I/O handling (including in-memory image support).
* **Chores**
  * Updated toolchain requirement to Zig 0.16.0; improved build/test targeting and repository ignore rules.
* **Tests**
  * Updated unit tests for revised disk/directory entry APIs and added additional autodetection/edge-case coverage.

### CODE REVIEW

File: src/disk_image.zig [hunk 1]
```diff
@@ -129,190 +207,308 @@ pub const DiskImage = struct {
...
+                            continue;
+                        } else {
+                            return dt;
+                        }
+                    },
+                    else => return dt,
+                }
             }
         }
         return null;
```
[the remark this answers is not in the document]
Comment (phatchman) [reply src/disk_image.zig:237]: File is passed by value here. It won't change the seek pos.

File: src/main.zig [hunk 2]
```diff
@@ -27,30 +27,18 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 // SOFTWARE.
 //
-var gpa: std.heap.GeneralPurposeAllocator(.{}) = .init;
-pub const global_allocator = gpa.allocator();
 
-// This arena lasts the lifetime of the application.
-pub var arena = std.heap.ArenaAllocator.init(global_allocator);
+// TODO: Errors not being shown for invalid options e.g. -q
+
```
[the remark this answers is not in the document]
Comment (phatchman) [reply src/main.zig:32]: it doesn;t the issue is the dependency for the cli is missing a flush. all errors have already been procssed at that point
